frontend/src/components/NavDropdown.tsx

```diff
@@ -0,0 +1,107 @@
+import clsx from "clsx";
+import { noop, toNumber } from "lodash";
+import { IndexType } from "src/types/generalTypes";
+
+import {
+  Dispatch,
+  JSX,
+  MouseEvent,
+  SetStateAction,
+  useEffect,
+  useState,
+} from "react";
+import { Menu, NavDropDownButton } from "@trussworks/react-uswds";
+
+interface NavDropdownProps {
+  activeNavDropdownIndex: IndexType;
+  index: number;
+  isCurrent: boolean;
+  linkText: string;
+  menuItems: JSX.Element[];
+  setActiveNavDropdownIndex: Dispatch<SetStateAction<IndexType>>;
+}
+
+export default function NavDropdown({
+  activeNavDropdownIndex,
+  index,
+  isCurrent,
+  linkText,
+  menuItems,
+  setActiveNavDropdownIndex,
+}: NavDropdownProps): JSX.Element {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  function eventHandler(e: MouseEvent) {
+    const dropdowns = document.getElementsByName("navDropDownButton");
+    let dropdownClicked = false;
+    for (const dropdown of dropdowns) {
+      if (dropdown.contains(e.target as Node)) {
+        dropdownClicked = true;
+      }
+    }
+
+    let targetId = null;
+    if (dropdownClicked) {
+      let targetNode: HTMLElement = e.target as HTMLElement;
+      if (targetNode.localName === "span") {
+        targetNode = targetNode.parentNode as HTMLElement;
+      }
+      const targetNodeClass: string = targetNode.className;
+      if (!targetNodeClass.includes("simpler-subnav-open")) {
+        targetId = toNumber(targetNode.id);
+      }
+    }
+    setActiveNavDropdownIndex(targetId);
+  }
+
+  function handleToggle(e: MouseEvent<HTMLButtonElement, MouseEvent>) {
+    const activeIndex: IndexType = isOpen ? null : index;
+    if (activeIndex) {
+      e.stopPropagation();
+      requestAnimationFrame(() =>
+        document.addEventListener(
+          "click",
+          // eslint-disable-next-line
+          // @ts-ignore
+          (ev: MouseEvent<Element, MouseEvent>): void => eventHandler(ev),
+          {
+            once: true,
+          },
+        ),
+      );
+    }
+    setActiveNavDropdownIndex(activeIndex);
+  }
+
+  useEffect(() => {
+    setIsOpen(activeNavDropdownIndex === index);
+  }, [activeNavDropdownIndex, index]);
+
+  return (
+    <>
+      <NavDropDownButton
+        id={index.toString()}
+        name="navDropDownButton"
+        label={linkText}
+        menuId={linkText}
+        isOpen={isOpen}
+        // eslint-disable-next-line
+        // @ts-ignore
+        onClick={(e: MouseEvent<HTMLButtonElement, MouseEvent>) =>
```

Comment:
I was able to work around the TS problems here
* no type assignments on the click handler
* handleToggle takes a MouseEvent
* closeNavs takes an Event